### PR TITLE
fix: calculation of nb_dimensions for approximated inverse

### PR DESCRIPTION
**Pull request by tas17**

### CODE REVIEW

File: saiph/inverse_transform.py [hunk 1]
```diff
@@ -37,7 +38,7 @@ def inverse_transform(
     random_gen = np.random.default_rng(model.seed)
 
     # Check dimension size regarding N
-    n_dimensions = len(model.dummy_categorical) + len(model.original_continuous)
+    n_dimensions = len(model.dummy_categorical) + len(model.original_continuous) - 1
```
Comment (jpetot) [inline saiph/inverse_transform.py:41]:
```suggestion
    n_dimensions = len(model.dummy_categorical) + len(model.original_continuous) - nb_variable_categorical 
```

**Review by jpetot — CHANGES_REQUESTED** [review]

File: saiph/inverse_transform.py (hunk 1 above)
Comment (jpetot) [reply saiph/inverse_transform.py:41]: could you create unit test of this fix?

File: saiph/inverse_transform.py [hunk 2]
```diff
@@ -37,7 +38,10 @@ def inverse_transform(
     random_gen = np.random.default_rng(model.seed)
 
     # Check dimension size regarding N
-    n_dimensions = len(model.dummy_categorical) + len(model.original_continuous)
+    n_dimensions = get_number_of_dimensions_from_categorical_and_continuous(
+        dummy_categorical=model.dummy_categorical,
+        original_continuous=model.original_continuous,
+    )
```
Comment (tas17) [inline saiph/inverse_transform.py:44]: Note: I was first using `model: Model` as the only input, but as it's a `@DataClass`, I could not generate a factory for it.
So I went to use that workaround instead

File: saiph/models.py [hunk 3]
```diff
@@ -69,3 +69,9 @@ class Model:
     dummies_col_prop: Optional[NDArray[np.float64]] = None  # MCA only
 
     seed: Optional[int] = None
+
+
+def get_number_of_dimensions_from_categorical_and_continuous(
+    *, dummy_categorical: List[Any], original_continuous: List[Any]
+) -> int:
+    return len(dummy_categorical) + len(original_continuous) - 1
```
Comment (jpetot) [inline saiph/models.py:77]:
```suggestion
    return len(dummy_categorical) + len(original_continuous) - nb_categorical_variable
```

**Review by jpetot — CHANGES_REQUESTED** [review]
Sorry if I wasn't clear enough for this point

**Pull request closed without merging**